### PR TITLE
Update filters interface

### PR DESCRIPTION
**Pull request by jamesgorrie**
Preparing for release of searching for books with online view.

* removes "Online" filter as it has to do with access, and we're going to start with surfacing books that are online, work on access, then offer filters. I didn't remove the logic to deal with this as it might be useful
* only shows filters when there are results

screencast-localhost-3000-2019 02 21-14-32-47

### CODE REVIEW

**Comment by Heesoomoon** [comment]
@jamesgorrie We can discuss this in person on Monday. I'm just documenting this so I don't forget. No need to reply.
Here are the relevant user stories for the search filters with the understanding that initial search functionality will be for digitised image collection (artworks, photos and more) and books.
- As a researcher, I want to see everything Wellcome has on 'keyword'. 
- As an image researcher, I want to only look at image collection results with the images relevant to my research topic.

The reason I recommended to remove the filters on the landing (/works) is so we don't overwhelm users with too many options up front. However, we may want to provide an option 'search images only' for the existing image researchers who are solely interested in our image collection. Let's follow up with @jennpb for her product strategy.

In terms of the result page, few things to follow up:
- The result should display all results by default. Then users should be able to narrow their search results by material types. Therefore, I would like to discuss the options (faceted nav) rather than solving it with the checkbox which poses some usability issues. 
- 'Images' as a work type filter may be too generic and unclear to the users. Need some team discussion.

I'll update #4098 Search Bar to capture the open issues and design considerations.

**Comment by jamesgorrie** [comment]
Just so I remember for Monday.

The question this is asking and partially answering is
__How do we become more than an image search?__

Two potential solutions that I can think of are:
• Open the whole API
• Open up parts of the API, i.e. online books first

Each option will have a set of things we need to get it into a testable state, and it might be good to go through what those are.

Reasons to be more than an image search:
* To be collecting data on how people search on more "real life data". Books was mentioned as a good candidate as it reflects a good chunk of our catalogue and searching patterns.
* To be able to move forward, assuming incrementally 

> faceted nav

[I thought checkboxes were and example of faceted nav](https://www.nngroup.com/articles/filters-vs-facets/), as you can have multiple selected vs radio buttons where you have to chose _an_ option e.g. only one type?

If it's the interface and we want it to be more  buttony<sup>*</sup>, that would should be easy to get done, there was a point in #4134 about design that @Heesoomoon raised in our meeting that hasn't been created or fleshed out.
e.g. screenshot 2019-02-22 at 16 33 38

<sup>*</sup> 😿Terrible wording, sorry.

File: catalogue/webapp/components/SearchForm/SearchForm.js [hunk 1]
```diff
@@ -190,74 +193,12 @@ const SearchForm = ({
...
-                    }
-                  }}
-                />
-              </fieldset>
               <p
                 className={classNames({
                   [font({ s: 'HNL4' })]: true,
                   'no-margin': true,
                   relative: true,
+                  [spacing({ s: 2 }, { margin: ['right'] })]: true,
```
Comment (davidpmccormick) [inline catalogue/webapp/components/SearchForm/SearchForm.js]: Does this win over `no-margin` above?

File: catalogue/webapp/components/SearchForm/SearchForm.js [hunk 2]
```diff
@@ -269,6 +210,55 @@ const SearchForm = ({
...
+                  style={{
+                    left: '1px',
+                  }}
+                >
+                  <legend
+                    className={classNames({
+                      'float-l': true,
+                      [font({ s: 'HNL4' })]: true,
+                    })}
+                    style={{ marginTop: '3px' }}
```
Comment (davidpmccormick) [inline catalogue/webapp/components/SearchForm/SearchForm.js]: One for _The Great Spacing Debate_

File: catalogue/webapp/components/SearchForm/SearchForm.js (hunk 1 above)
Comment (jamesgorrie) [reply catalogue/webapp/components/SearchForm/SearchForm.js]: Correcto - maybe not the way to express it, but it works.

File: catalogue/webapp/components/SearchForm/SearchForm.js (hunk 2 above)
Comment (jamesgorrie) [reply catalogue/webapp/components/SearchForm/SearchForm.js]: What about, _The great debate of space_?
Or _The great DebAce_?

File: catalogue/webapp/components/SearchForm/SearchForm.js (hunk 1 above)
Comment (davidpmccormick) [reply catalogue/webapp/components/SearchForm/SearchForm.js]: Strange – I'm sure I've encountered a situation where `no-margin` wins.

Comment (jamesgorrie) [reply catalogue/webapp/components/SearchForm/SearchForm.js]: screenshot 2019-02-26 at 11 14 54

Correct, now fixed.

**Pull request merged**